lib/web_browser.pm

```diff
@@ -0,0 +1,84 @@
+# Copyright (C) 2019 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+# Summary: common functions for FIPS web browser tests
+# Maintainer: llzhao <llzhao@suse.com>
+
+package web_browser;
+
+use base Exporter;
+use Exporter;
+use strict;
+use warnings;
+use testapi;
+use utils;
+use version_utils 'is_sle';
+use registration qw(add_suseconnect_product register_product);
+
+our @EXPORT = qw(
+  setup_web_browser_env
+  run_web_browser_text_based
+);
+
+# Setup test envs: register PackageHub and check FIPS pattern installed
+sub setup_web_browser_env {
+    my $ret = 0;
+
+    # Check the register status for reference
+    script_run("SUSEConnect --list-extension | grep 'SUSE Package Hub' | grep '(Activated)'");
+    $ret = script_run("output=`SUSEConnect -s | grep 'PackageHub'`; echo \${output##*PackageHub} | cut -d ':' -f4 | grep 'Registered'");
```

Comment:
What if there are other products right after the PackageHub? Please find a reasonable solution to parse the json output.

Reply:
I have considered this situation, the code has no lack ("cut -d ':' -f4" can work on this situation).
Thanks and really good checking.